packages/compose/src/hooks/use-disabled/index.js

```diff
@@ -50,56 +54,54 @@ const DISABLED_ELIGIBLE_NODE_NAMES = [
  * ```
  */
 export default function useDisabled() {
-	/** @type {import('react').RefObject<HTMLElement>} */
-	const node = useRef( null );
+	return useRefEffect( ( node ) => {
+		const disable = () => {
+			node.style.setProperty( 'user-select', 'none' );
+			node.style.setProperty( '-webkit-user-select', 'none' );
```

Comment:
This change means you can select text inside a disabled element. For instance, I noticed that when double clicking a paragraph it would get selected even it had a "group" parent, that's because the "useSelectionObserver" hook would have triggered anyway. Preventing browser selection solves this.

Reply:
Do you mean whole block or just the text?

I would expect text to be selectable even if component is disabled. Mostly for copy/pasting.

Reply:
I think disabling text selection for previews is okay, but will be more problematic UX decision for blocks in the editor.

There are few blocks that use Disabled component.

Reply:
The problem is that there's no way to know that a block is "disabled", it's not something in state. so there's no way to disabled useSelectionObserver when a block is disabled. So I'm not sure how to solve that otherwise.

Reply:
Good point 😅 I’m okay shipping the fix like this and keeping eye on the feedback.

Reply:
That's a tricky one. What problem does this cause in practice?

Reply:
Right now in trunk "nothing" but I'm considering using useDisabled to disable blocks for several things:

 - exploded mode
 - block overlay (template parts and reusable blocks)
 - potentially locking as well

The problem is that useSelectionObserver triggers, clicking a block will select it even if it's a child of a disabled block.